pkg/envoy/xds/server.go

```diff
@@ -254,6 +263,23 @@ func (s *Server) processRequestStream(ctx context.Context, streamLog *logrus.Ent
 
 	nodeIP := ""
 
+	if s.restorerPromise != nil {
+		restorer, err := s.restorerPromise.Await(context.Background())
+		if err != nil {
+			return err
+		}
+
+		if restorer != nil {
+			streamLog.Debug("Waiting for endpoint restoration before serving resources...")
+			restorer.WaitForEndpointRestore(ctx)
+			for typeURL, ackObserver := range s.ackObservers {
```

Comment:
question: should we bailt out early if `ctx` has been cancelled? (this is likely moot and only relevant for tests)

Reply:
Presumably we could pass the `ctx` to `Await()` instead of the `context.Background()` to achieve this?

Reply:
I'll changed Await to take `ctx` rather than `context.Background()`